collector/runtime-benchmarks/bufreader/src/main.rs

```diff
@@ -1,34 +1,36 @@
-use benchlib::benchmark::{black_box, BenchmarkSuite};
-use snap::{read::FrameDecoder, write::FrameEncoder};
 use std::io::{BufRead, BufReader, Write};
 
+use snap::{read::FrameDecoder, write::FrameEncoder};
+
+use benchlib::benchmark::{benchmark_suite, black_box};
+use benchlib::define_benchmark;
+
 const BYTES: usize = 64 * 1024 * 1024;
 
 fn main() {
-    let mut suite = BenchmarkSuite::new();
-
     // Inspired by https://github.com/rust-lang/rust/issues/102727
     // The pattern we want is a BufReader which wraps a Read impl where one Read::read call will
     // never fill the whole BufReader buffer.
-    suite.register("bufreader-snappy", || {
-        let data = vec![0u8; BYTES];
-        move || {
-            let mut compressed = Vec::new();
-            FrameEncoder::new(&mut compressed)
-                .write_all(&data[..])
-                .unwrap();
-            let mut reader = BufReader::with_capacity(BYTES, FrameDecoder::new(&compressed[..]));
+    benchmark_suite(|suite| {
+        define_benchmark!(suite, bufreader_snappy, {
```

Comment:
This pattern of `define_benchmark!` within `benchmark_suite` occurs every time. Can the latter be moved into the macro?

Reply:
Hmm, well there is just one benchmark suite, for which you can define_benchmark! multiple benchmarks. It's just that right now there is only a single benchmark in each suite, but there should be more, once we add them. So I need some unifying element (the suite) to collect the benchmarks together. Unless I use e.g. thread locals or a global variable for this.